hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -83,27 +84,34 @@ public HollowProducer(
         this.validator = validator;
         this.announcer = announcer;
 
-        writeEngine = new HollowWriteStateEngine();
-        objectMapper = new HollowObjectMapper(writeEngine);
+        writeEngineWrapper = new WriteEngineWrapper();
         versionMinter = new VersionMinterWithCounter();
         listeners = new ListenerSupport();
         readStates = ReadStateHelper.newDeltaChain();
     }
 
     public void initializeDataModel(Class<?>...classes) {
         long start = currentTimeMillis();
-        for(Class<?> c : classes)
-            objectMapper.initializeTypeState(c);
+        writeEngineWrapper.initializeDataModel(classes);
         listeners.fireProducerInit(currentTimeMillis() - start);
     }
-    
+
     public void initializeDataModel(HollowSchema... schemas) {
         long start = currentTimeMillis();
-        HollowWriteStateCreator.populateStateEngineWithTypeWriteStates(writeEngine, Arrays.asList(schemas));
+        writeEngineWrapper.initializeDataModel(schemas);
         listeners.fireProducerInit(currentTimeMillis() - start);
     }
 
+    public void setRestoreFailureFatal(boolean isFatal) {
```

Comment:
I think that the behavior of the system when restore fails should be defined by the caller to restore, rather than have to configure it here.